python/eups/Eups.py

```diff
@@ -3041,10 +3041,32 @@ def getDependentProducts(self, topProduct, setup=False, shouldRaise=False,
         if not prodtbl:
             return dependentProducts
 
-        for product, optional, recursionDepth in prodtbl.dependencies(self, recursive=True, recursionDepth=1,
-                                                                      followExact=followExact,
-                                                                      productDictionary=productDictionary,
-                                                                      requiredVersions=requiredVersions):
+        productDict = {}
+        dependencies = prodtbl.dependencies(self, recursive=True, recursionDepth=1,
```

Comment:
Well, we need to get the dependencies of the default product to see if the top-level product is in it.  In order to know which version of the default product, we need to get the dependencies of the top-level product.  And then if (and only if) the top-level product is in the default product dependency list, we need a new dependency list for the top-level product.  I don't see a way around this, though @RobertLuptonTheGood may know one.
In the usual case, `dependencies()` is called only twice, and one of those is on the default product, which should be fairly quick (and I *think* the table files and all are already read in, so it's "just" CPU), so I don't *think* this is a terrible performance degradation.